Makefile

```diff
@@ -55,6 +55,9 @@ doccheck: generate
 	@cd docs; for doc in *.md; do if [ "$$doc" != "README.md" ] && ! grep -q "$$doc" *.md; then echo "ERROR: No link to documentation file $${doc} detected"; exit 1; fi; done
 	@echo OK
 
+docs:
```

Comment:
As above, just adding a check for docs under doccheck would be great, something like this should due it in theory:

```
doccheck: generate docs
	@echo "- Checking if the generated documentation is up to date..."
	@git diff --exit-code
	@echo "- Checking if the documentation is in sync with the code..."
	@grep -hoE -d skip '\| kube_[^ |]+' docs/* --exclude=README.md | sed -E 's/\| //g' | sort -u > documented_metrics
	@find internal/store -type f -not -name '*_test.go' -exec sed -nE 's/.*"(kube_[^"]+)".*/\1/p' {} \; | sort -u > 
```

Reply:
Why was this resolved? I don't see this comment being applied. I agree with @lilic that this is how we should do this.

Reply:
@brancz I have added below comments to explain the things what we have now and got resolved this temporarily ,excuse me. Will keep this open.  Planning to add a seperate step for checking check dynamically added docs.

Reply:
@brancz @lilic  Please find the new updates

1. Created a separate location(./docs/dynamic-docs) for keeping the automatically generated docs because of the conflicts with existing docs while verifying.  As for testing, we did created a doc automtically using the package 'docs' for configmap-metrics as an example: https://github.com/kubernetes/kube-state-metrics/pull/1379/files#diff-e4acaed3adc42bd86d1d1ff96b032645cae2e31b6a6ebdf3dd8edc88288ae98e . If we remove the existing docs in the repo, we will be missing some informations which is not getting from the `family generator` type as already discussed with @lilic. I suggest to keep this as separate until the automatic docs generation will be stable. 

2. I have added new stage called `ci-generate-docs` for automatically creating docs files and also checking dynamically created docs file as discussed. Now we are having a new target called `doccheck-dynamicdocs`.

Any other suggestions, kindly please let me know

Reply:
@lilic If we are good to move with current structure, I will fix current issue with pipeline while validating.

Reply:
Not sure we need to name these dynamic docs directory. Having it be just in docs is better also for checking that the diff is not too large but also ease of discovery for our end users.

What was the issue with the ConfigMap vs other metrics?

Reply:
@lilic  We are actually generating docs from what we are getting from family generator. We are still having certain fields which needs to be added either manually or we have to incorporate this details to the family generator--> let me know your thought on this. As you know unless we have this details added, we cannot overwrite the existing docs instead we want to keep the docs and that is why I created under a separate directory.

Reply:
Yes, I understand we are generating the docs from the struct, but I would prefer to have these details in the code itself and then generate the docs based on them. e.g. status/stability be a new field in the metricsfamily struct. Labels we already have in code not sure how we can get those but they are there right now right?

@brancz I think if we add the missing fields like stability API which we wanted to add anyways.

I would prefer if we do this properly by generating all our docs so we don't have to keep updating in two places and having code be the source of all truth.

that is indeed a bit more of work and if its too much for you right now that is okay, but having two directories of docs can be confusing for our users rather than clarifying.

Reply:
@lilic  

This are the few suggestions
1. Standardize the family-generator declaration for all the metrics, add the missing fields to the existing declarations.
2. After the review, package 'docs' can be added or merged 
3. Or Keep the PR opened unless we fixed (1) so as a fully automated doc can be generated

Reply:
Yes thanks, I agree with keeping this PR open, while at the same time working on first (1) point of adding the missing things to our code base that we currently only have in our docs, this will allow us to generate the docs from our code fully. @JESWINKNINAN are you willing to work on the first point, otherwise could you open a new issue for this and we can mark it as help wanted.

cc @brancz @tariq1890 @mrueg